lib/segment/src/index/field_index/map_index/immutable_map_index.rs

```diff
@@ -137,7 +133,16 @@ impl<N: Hash + Eq + Clone + Display + FromStr + Default> ImmutableMapIndex<N> {
                 // update db
                 let key = MapIndex::encode_db_record(value, idx);
                 self.db_wrapper.remove(key)?;
+                removed_values_count += 1;
+            }
+
+            if removed_values_count > 0 {
+                self.indexed_points -= 1;
             }
+            self.values_count = self
+                .values_count
+                .checked_div(removed_values_count)
+                .unwrap_or_default();
```

Comment:
I don't think this matches with `self.values_count -= removed_values.len();` from before. Is that intentional?

Reply:
thanks! it's a bug